autosklearn/metalearning/metafeatures/metafeatures.py

```diff
@@ -1082,11 +1082,19 @@ def calculate_all_metafeatures(X, y, categorical, dataset_name, logger,
                 # TODO make sure this is done as efficient as possible (no copy for
                 # sparse matrices because of wrong sparse format)
                 sparse = scipy.sparse.issparse(X)
+
+                feat_type = {key: 'categorical' if value else 'numerical'
+                             for key, value in categorical.items()}
+
+                # Todo make this more cohesive to the overall structure (quick bug fix)
```

Comment:
```suggestion
                # TODO make this more cohesive to the overall structure (quick bug fix)
```

Reply:
Could you please also make a todo list with todos to be done after this PR is merged (we could make a new issue for these) so that we keep track of what still needs to be done?